packages/user-mgnt/src/features/createUser/service.test.ts

```diff
@@ -35,25 +34,29 @@ const token = jwt.sign(
 
 describe('Username generation', () => {
   it('generates valid username with given and family names', async () => {
-    mockingoose(UsernameRecord).toReturn(null, 'findOne')
-    mockingoose(UsernameRecord).toReturn(null, 'save')
+    mockingoose(User).toReturn(null, 'findOne')
     const names = [{ given: ['John', 'Evan'], family: 'Doe', use: 'en' }]
     const username = await generateUsername(names)
     expect(username).toBe('je.doe')
     mockingoose.resetAll()
   })
 
   it('generates valid username with given and family names with appended number', async () => {
-    mockingoose(UsernameRecord).toReturn(mockUsernameRecord, 'findOne')
-    mockingoose(UsernameRecord).toReturn(null, 'update')
+    const finderMock = (query: any) => {
```

Comment:
```suggestion
    const finderMock = (query: {getQuery: () => Record<string, unknown>}) => {
```
Would this type help perhaps?